src/legacy/core_plugins/expressions/public/np_ready/public/expression_renderer.tsx

```diff
@@ -70,5 +93,37 @@ export const createRenderer = (loader: IExpressionLoader): ExpressionRenderer =>
     mountpoint.current,
   ]);
 
-  return <div {...dataAttrs} className={className} ref={mountpoint} />;
+  useEffect(() => {
+    // We only want a clean up to run when the entire component is unloaded, not on every render
+    return function cleanup() {
+      if (handlerRef.current) {
+        handlerRef.current.destroy();
+        handlerRef.current = null;
+      }
+    };
+  }, []);
+
+  const classes = classNames('expExpressionRenderer', className, {
+    'expExpressionRenderer-isLoading': state.isLoading,
+    'expExpressionRenderer-hasError': state.hasError,
+  });
+
+  return (
+    <div {...dataAttrs} className={classes}>
+      {state.isLoading ? (
+        <>
+          <EuiProgress size="xs" color="accent" position="absolute" />
+          <EuiLoadingChart mono size="l" />
+        </>
+      ) : null}
```

Comment:
@wylieconlon Sorry my comment was probably not super clear. What I was trying to achieve was have the EuiLoadingChart be displayed **if there is not already a chart displaying**, like when adding your first field. **Then**, if the chart starts "loading" again, like adding a split field, it should keep displaying the old chart, but use the EuiProgress, at the top to indicate the chart is re-loading.

Both loaders should not be displayed at the same time.